api/python/provisioner/__main__.py

```diff
@@ -169,7 +172,10 @@ def _set_logging(output_type, log_args=None, other_args=None):
                 else:
                     filename = _generate_logfile_filename(log_args.cmd)
                 setattr(log_args, filename_attr, str(filename))
-            log_args.update_handlers()
+
+    if (cmd in config.LOG_SUPPRESSION_CMDS or
```

Comment:
what if we can accumulate all changes in some key-value manner and apply them all here using `setattr` so we won't need to check these conditionals one more time?

Reply:
Refactoring this. But python needs to support `finally` for if-else conditional :slightly_smiling_face:

Reply:
possibly... but I meant that we can:
```
log_changes = {}
if(cond1):
   if (conf2):
      log_changes[attr1] = val1

...

if (cond3):
    log_changes[attr2] =- val2

if log_changes:
    for lattr, lval in ...:
        setatt(log_args, lattr, lval)
    log_args.update_handlers()  
```